include/odp/api/spec/packet.h

```diff
@@ -71,6 +71,51 @@ extern "C" {
   * Packet is red
   */
 
+/**
+ * Protocol
+ */
+typedef enum odp_proto_t {
+	/** No protocol defined */
+	ODP_PROTO_NONE = 0,
+
+	/** Ethernet (including VLAN) */
+	ODP_PROTO_ETH,
+
+	/** IP (including IPv4 and IPv6) */
+	ODP_PROTO_IP,
+
+	/** UDP */
```

Comment:
What is the usecase for parsing a packet starting from L4 header? Also there are several (lots) of other L4 protocols. Do we want to support them all here?

Reply:
@lumag IPsec operating in transport mode is, I'd imagine, the main use case here.

Reply:
I felt easier to reparse both L3 and L4 headers in IPsec case, especially since transport mode ESP can en/decrypt some of L3 headers in IPv6 case.

Reply:
It's there to enable application to call parsing in parts - e.g. first up to IP and then continue from L4. But since IP and transport protocols are tied together with pseudo headers, it's cleaner to remove L4 as a starting point.